src/ctapipe_io_nectarcam/__init__.py

```diff
@@ -891,29 +1044,52 @@ def fill_r0r1_camera_container(self, zfits_event):
         in the file.
         Missing or broken pixels are filled using maxval of the waveform dtype.
         """
-        n_pixels = self.camera_config.num_pixels
-        n_samples = self.camera_config.num_samples
-        expected_pixels = self.camera_config.expected_pixels_id
+
+        #nectarcam_service contain the info from camera_config but in a "uniform" way between old and v6 data
+        n_pixels        = self.nectarcam_service.num_pixels  if self.pre_v6_data else zfits_event.num_pixels
+        n_samples       = self.nectarcam_service.num_samples if self.pre_v6_data else zfits_event.num_samples
+        expected_pixels = self.nectarcam_service.pixel_ids
 
         has_low_gain = (zfits_event.pixel_status & PixelStatus.LOW_GAIN_STORED).astype(bool)
         has_high_gain = (zfits_event.pixel_status & PixelStatus.HIGH_GAIN_STORED).astype(bool)
         not_broken = (has_low_gain | has_high_gain).astype(bool)
+        all_broken = np.count_nonzero(has_low_gain) == 0 and np.count_nonzero(has_high_gain) == 0
+        
+        # all_broken is likely because of the issue : 
+        # https://redmine.cta-observatory.org/issues/52457 
+        # so it could be a way to discriminate
 
         # broken pixels have both false, so gain selected means checking
         # if there are any pixels where exactly one of high or low gain is stored
         gain_selected = np.any(has_low_gain != has_high_gain)
 
         # fill value for broken pixels
-        dtype = zfits_event.waveform.dtype
-        fill = np.iinfo(dtype).max
+        
+        if self.pre_v6_data:
+            dtype = zfits_event.waveform.dtype
+            fill = np.iinfo(dtype).max
+        else:
+            dtype = np.float32 
+            fill = np.finfo(dtype).max
+        # VIM : We received uint16 data from the EVB, 
+        # but because of the scaling in v6 we gets back to a float, so let's use directly floats
+
         # we assume that either all pixels are gain selected or none
         # only broken pixels are allowed to be missing completely
+
+        # if all_broken:
```

Comment:
Is this code meant to be added to the PR before merging or should it be cleaned up?